components/raftstore/src/store/worker/split_check.rs

```diff
@@ -196,12 +195,46 @@ where
             "region_id" => region_id,
             "start_key" => log_wrappers::Value::key(&start_key),
             "end_key" => log_wrappers::Value::key(&end_key),
+            "policy" => ?policy,
         );
         CHECK_SPILT_COUNTER.all.inc();
-
         let mut host =
             self.coprocessor
                 .new_split_checker_host(region, &self.engine, auto_split, policy);
+        if host.policy() == CheckPolicy::Approximate && host.enable_region_bucket() {
+            let mut bucket_keys = match host.approximate_bucket_keys(region, &self.engine) {
+                Ok(keys) => keys
+                    .into_iter()
+                    .map(|k| keys::origin_key(&k).to_vec())
+                    .collect(),
+                Err(e) => {
+                    error!(%e;
+                        "failed to get approximate bucket key";
+                        "region_id" => region_id,
+                    );
+                    vec![]
+                }
+            };
+            info!(
+                "starting approximate_bucket_keys {}, bucket {:?}",
+                bucket_keys.len(),
+                bucket_keys
+            );
+            if !bucket_keys.is_empty() {
```

Comment:
Should we initialize the buckets with `start_key` and `end_key` if `bucket_keys` is empty?

Reply:
i think we should. Because the region size could be shrunk and we should update the buckets upon the shrink.